### PR TITLE
Fix condition simplifier

### PR DESCRIPTION
**Pull request by ana-pantilie**
Part of #2508

- before simplifying each part of the condition with the assumption that the other parts are true, we simplify the whole predicate as we did before
- reverts change to unit test introduced in 9460590a964dbb25464f452094c0461635bf62de

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

**Comment by ttuegel** [comment]
Let's open new issues for any tests that fail with both the `master` version of the backend and this version of the backend using the same version of the frontend used in `evm-semantics`. For all the tests that pass on `master`, we should just make sure that this pull request doesn't create an additional regression.

**Comment by ana-pantilie** [comment]
@ttuegel Here is the data from the test runs, as we discussed in the meeting. I didn't generate the percentages, so I might have made mistakes, but the rest of the data should be accurate.

```
               Test name                              | Time PR | Time master |  Percentage  | Productivity on master
----------------------------------------------------------------------------------------------------------------------
tests/specs/functional/storageRoot-spec.k             |  1m  8s |    1m 25s   |  20% faster  |             - 
----------------------------------------------------------------------------------------------------------------------
tests/specs/functional/lemmas-spec.k                  |   55s   |    1m  3s   |  13% faster  |             - 
----------------------------------------------------------------------------------------------------------------------
tests/specs/examples/sum-to-n-spec.k                  |   2m    |    1m 40s   |  20% slower  |             - 
----------------------------------------------------------------------------------------------------------------------
tests/specs/functional/infinite-gas-spec.k            |   43s   |     47s     |   8% faster  |             - 
----------------------------------------------------------------------------------------------------------------------
tests/specs/functional/merkle-spec.k                  |   36s   |     38s     |   5% faster  |             - 
----------------------------------------------------------------------------------------------------------------------
tests/specs/functional/lemmas-no-smt-spec.k           |   39s   |     42s     |   7% faster  |             - 
----------------------------------------------------------------------------------------------------------------------
tests/specs/benchmarks/storagevar00-spec.k            |  7m 43s |    8m 10s   |   3% faster  |             - 
----------------------------------------------------------------------------------------------------------------------
tests/specs/benchmarks/overflow00-overflow-spec.k     | 11m 33s |   10m 11s   |  13% slower  |            90%
----------------------------------------------------------------------------------------------------------------------
tests/specs/benchmarks/storagevar03-spec.k            |  6m 36s |    6m  7s   |   8% slower  |            86%
----------------------------------------------------------------------------------------------------------------------
tests/specs/benchmarks/storagevar02-overflow-spec.k   | 11m  3s |   10m  9s   |   9% slower  |            86%
----------------------------------------------------------------------------------------------------------------------
tests/specs/benchmarks/storagevar02-nooverflow-spec.k | 11m 52s |   10m 54s   |   9% slower  |             -
----------------------------------------------------------------------------------------------------------------------
tests/specs/benchmarks/storagevar01-spec.k            | 10m 27s |    9m 28s   |  10% slower  |            84%
----------------------------------------------------------------------------------------------------------------------
tests/specs/erc20/hkg/totalSupply-spec.k              |  2m 46s |    2m 46s   |       -      |             -
----------------------------------------------------------------------------------------------------------------------
tests/specs/erc20/ds/totalSupply-spec.k               | 55m  5s |   59m 18s   |   7% faster  |             - 
```

**Comment by ttuegel** [comment]
I think that's fine. We see only small changes on the long-running tests. Those short tests have sudden speed-ups and slow-downs often, I'm not worried much about that. And anyway, we are working on the performance of all these tests right away.

File: kore/src/Kore/Step/Simplification/Condition.hs [hunk 1]
```diff
@@ -84,7 +84,11 @@ simplify SubstitutionSimplifier{simplifySubstitution} sideCondition =
     worker Conditional{term, predicate, substitution} = do
         let substitution' = Substitution.toMap substitution
             predicate' = Predicate.substitute substitution' predicate
-        simplified <- simplifyPredicates sideCondition predicate'
+        partiallySimplified <- simplifyPredicate sideCondition predicate'
+        simplified <-
+            simplifyPredicates
+                sideCondition
+                (from @_ @(Predicate _) partiallySimplified)
```
Comment (ttuegel) [inline kore/src/Kore/Step/Simplification/Condition.hs:91]:
```suggestion

        simplified <-
            simplifyPredicate sideCondition predicate'
            >>= simplifyPredicates sideCondition . from @_ @(Predicate _)
```

**Review by ttuegel — APPROVED** [review]
I left one minor suggestion, but otherwise it looks good!

**Pull request merged**